pkg/operator/resource_reconciler.go

```diff
@@ -261,7 +265,11 @@ func (rr *ResourceReconciler) OnAdd(obj interface{}, _ bool) {
 
 // OnUpdate implements the cache.ResourceEventHandler interface.
 func (rr *ResourceReconciler) OnUpdate(old, cur interface{}) {
-	if _, ok := cur.(*appsv1.StatefulSet); ok {
+	switch cur.(type) {
```

Comment:
(nit) you could use the same `switch v := obj.(type)` trick and pass `v` to the function.

Reply:
How? Here we don't have `obj` interface, we have two `old` and `cur` interface :thinking:

Reply:
I think simon had something like this https://github.com/prometheus-operator/prometheus-operator/pull/6787#discussion_r1718719044 in mind